### PR TITLE
basic/linux: update kernel headers from v6.10-rc3

### PR DESCRIPTION
**Pull request by yuwata**
This also
- merges basic/linux and shared/linux,
- moves BPF_JUMP_A() to basic/missing_bpf.h,
- copies from usrspace kernel headers directory generated by 'make headers', rather than copying from kernel tree,
- copies const.h into our tree to reduce change in ethtool.h,
- copies auto_fs.h into our tree to reduce change in auto_dev-ioctl.h,
- adds an entry for bcachefs to filesystems-gperf.gperf.

### CODE REVIEW

File: src/basic/linux/update.sh [hunk 1]
```diff
@@ -3,9 +3,31 @@
...
+            sed -r -i 's/^(#include <linux\/fs\.h>)/#if WANT_LINUX_FS_H\n\1\n#endif/' "$i"
+            ;;
+        auto_dev-ioctl.h)
+            sed -r -i '/^#define[[:space:]]+AUTOFS_DEV_IOCTL_VERSION_MINOR/ s/[0-9]+$/0/' "$i"
+            ;;
+        ethtool.h)
+            sed -r -i '/return (ep->speed_hi << 16) | ep->speed;/ s/return .*;/return ((__u32) ep->speed_hi << 16) | (__u32) ep->speed;/' "$i"
+            ;;
+        dm-ioctl.h)
+            sed -r -i '/#define[[:space:]]+DM_VERSION_MINOR/ s/[0-9]+/27/' "$i"
```
Comment (YHNdnzj) [inline src/basic/linux/update.sh]:
```suggestion
            sed -r -i '/^#define[[:space:]]+DM_VERSION_MINOR/ s/[0-9]+$/27/' "$i"
```

To be consistent with auto_dev-ioctl.h above?

File: src/core/exec-invoke.c [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
+#include <linux/fs.h>
```
Comment (YHNdnzj) [inline src/core/exec-invoke.c]: Hmm, why is the comment regarding sys/mount.h dropped?

File: src/basic/linux/btrfs_tree.h [hunk 3]
```diff
@@ -1094,14 +1090,14 @@ struct btrfs_file_extent_item {
 	__u8 encryption;
 	__le16 other_encoding; /* spare for later use */
 
-	/* are we inline data or a real extent? */
+	/* are we __inline__ data or a real extent? */
```
Comment (qhill) [inline src/basic/linux/btrfs_tree.h:1093]: I don't think the change to the comment is intended.

File: src/basic/linux/btrfs_tree.h [hunk 4]
```diff
@@ -1094,14 +1090,14 @@ struct btrfs_file_extent_item {
...
 	__u8 type;
 
 	/*
 	 * disk space consumed by the extent, checksum blocks are included
 	 * in these numbers
 	 *
-	 * At this offset in the structure, the inline extent data start.
+	 * At this offset in the structure, the __inline__ extent data start.
```
Comment (qhill) [inline src/basic/linux/btrfs_tree.h:1100]: Also here I don't think the change to the comment is intended.

File: src/basic/linux/btrfs_tree.h [hunk 5]
```diff
@@ -220,18 +216,18 @@
 #define BTRFS_METADATA_ITEM_KEY	169
 
 /*
- * Special inline ref key which stores the id of the subvolume which originally
+ * Special __inline__ ref key which stores the id of the subvolume which originally
```
Comment (qhill) [inline src/basic/linux/btrfs_tree.h:219]: And here.

File: src/basic/linux/btrfs_tree.h [hunk 6]
```diff
@@ -220,18 +216,18 @@
...
  * created the extent. This subvolume owns the extent permanently from the
  * perspective of simple quotas. Needed to know which subvolume to free quota
  * usage from when the extent is deleted.
  *
- * Stored as an inline ref rather to avoid wasting space on a separate item on
- * top of the existing extent item. However, unlike the other inline refs,
+ * Stored as an __inline__ ref rather to avoid wasting space on a separate item on
+ * top of the existing extent item. However, unlike the other __inline__ refs,
```
Comment (qhill) [inline src/basic/linux/btrfs_tree.h:225]: Also here.

File: src/basic/linux/btrfs_tree.h [hunk 7]
```diff
@@ -220,18 +216,18 @@
...
  * there is one one owner ref per extent rather than one per extent.
  *
- * Because of this, it goes at the front of the list of inline refs, and thus
- * must have a lower type value than any other inline ref type (to satisfy the
- * disk format rule that inline refs have non-decreasing type).
+ * Because of this, it goes at the front of the list of __inline__ refs, and thus
+ * must have a lower type value than any other __inline__ ref type (to satisfy the
+ * disk format rule that __inline__ refs have non-decreasing type).
```
Comment (qhill) [inline src/basic/linux/btrfs_tree.h:230]: And here.

File: src/basic/linux/btrfs_tree.h (hunk 3 above)
Comment (yuwata) [reply src/basic/linux/btrfs_tree.h:1093]: This and below are all intentional. Such spurious replacement is caused by build script of the kernel.
Previously, we copied kernel headers from include/uapi/linux, but now they are from usr/include/linux.
The build script of the kernel simply replace `inline` with `__inline__`, even the term is in comment.
Please report that to the kernel team.

File: src/core/automount.c [hunk 8]
```diff
@@ -4,7 +4,6 @@
 #include <fcntl.h>
 #include <limits.h>
 #include <linux/auto_dev-ioctl.h>
-#include <linux/auto_fs4.h>
```
Comment (poettering) [inline src/core/automount.c:7]: hmm, the merge of this header with auto_fs.h was done in 4.18. We currently have 4.15 as "recommended baseline", and 3.15 as "minimum baseline"

**Comment by poettering** [comment]
this generally looks good to me, except for the last patch, see above

File: src/core/automount.c (hunk 8 above)
Comment (yuwata) [reply src/core/automount.c:7]: Now the header auto_fs.h is imported from the latest kernel tree. Hence, it is not necessary to consider backward compatibility about the header.

**Pull request merged**